src/borda_clj/client.clj

```diff
@@ -14,21 +19,25 @@
       (if (< (count measurements) max-buffer-size)
         (assoc measurements key values) ; space available, add to buffer
         (do (print "borda buffer full, discarding measurement" dimensions values)
-            measurements))))) ; buffer full
+            (update measurements discard_key (partial merge-with +) {:success_count 1})))))) ; buffer full
```

Comment:
This is nice for symmetry with how we're updating this elsewhere, but a shorter version would be:

```clojure
(update-in measurements [discard_key :success_count] inc)
```

`update-in` is like a deep version of `update`, where the vector in the second argument is a path to the element you want to update.

Digression: This works well for not too complicated / deeply nested structures.  For the hairier ones, there's [this library](https://github.com/nathanmarz/specter).  I was very excited when I learned about it, and so seemed Uaalto when I told him about it, but to be honest we've never had a strong motivation to use it yet.

Reply:
Spectre's navigator does indeed look handy!